src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -82,27 +84,35 @@
       def[1] = def[1].trim();
       String name = def[0];
       ColumnType type;
-      if ("int".equals(def[1].toLowerCase())) {
+      if ("int".equals(def[1].toLowerCase(US))) {
         type = ColumnType.INT;
-      } else if ("string".equals(def[1].toLowerCase())) {
+      } else if ("string".equals(def[1].toLowerCase(US))) {
         type = ColumnType.STRING;
-      } else if ("timestamp".equals(def[1].toLowerCase())) {
+      } else if ("timestamp".equals(def[1].toLowerCase(US))) {
         type = ColumnType.TIMESTAMP;
-      } else if ("date".equals(def[1].toLowerCase())) {
+      } else if ("date".equals(def[1].toLowerCase(US))) {
         type = ColumnType.DATE;
-      } else if ("time".equals(def[1].toLowerCase())) {
+      } else if ("time".equals(def[1].toLowerCase(US))) {
         type = ColumnType.TIME;
-      } else if ("long".equals(def[1].toLowerCase())) {
+      } else if ("long".equals(def[1].toLowerCase(US))) {
         type = ColumnType.LONG;
       } else {
         String errmsg = "Invalid UniqueKey type '" + def[1] + "'. Valid"
             + " types are: int, string, timestamp, date, time, and long.";
         throw new InvalidConfigurationException(errmsg);
       }
-      if (tmpTypes.containsKey(name)) {
+      if (tmpNames.contains(name)) {
         String errmsg = "Invalid db.uniqueKey configuration: key name '"
             + name + "' was repeated.";
         throw new InvalidConfigurationException(errmsg);
+      } else if (tmpTypes.containsKey(name) && tmpTypes.get(name) != type) {
+        // The ukDecls contain two keys that differ only in case but are of
+        // different types. Force a case-sensitive type lookup.
+        Map<String, ColumnType> caseSensitiveTypes = new TreeMap<>();
```

Comment:
OK, we're inside a for loop here. This is hurting my head a little. So you're going to swap a partial tmpTypes into new case-sensitive mapping, and then continue building the types. Maybe a comment to that effect: "Replace the partial map with a case-sensitive copy."